tests/unit/test_resolver.py

```diff
@@ -172,43 +172,69 @@ def _resolve_packages(request: Request) -> RequestOutput:
 
 
 @pytest.mark.parametrize(
-    "flags",
```

Comment:
Unit tests (even new ones, but most importantly modifying existing ones) need to be introduced **with** the code that they're testing, otherwise it might cause breakages due t bugs in the original code during git bisect and a subsequent `pytest` call, that's the general rule of thumb.
The exception are integration tests because during incremental development you may not have much to integration test with the code in development yet and so those are introduced always when that condition is no longer true.

Reply:
I won't dive in further review of the test, because based on mu functional comments I expect the tests to be heavily impacted and hence needing an update.